src/Faker/Generator.php

```diff
@@ -205,6 +205,55 @@ class Generator
     protected $providers = [];
     protected $formatters = [];
 
+    /**
+     * @var UniqueGenerator
+     */
+    private $uniqueGenerator;
+
+    public function __construct()
+    {
+        $this->uniqueGenerator = new UniqueGenerator($this);
+    }
+
+    /**
+     * With the unique generator you are guaranteed to never get the same two
+     * values.
+     *
+     * @return self The UniqueGenerator is a proxy
+     */
+    public function withUnique()
```

Comment:
```suggestion
    public function withUnique(): self
```

Reply:
But this is technically not true. (And also why psalm is complaining). 

The `UniqueGenerator` and other generators will implement the magic method `__call()`. They can be used and treated as they are a `Generator` but they are not really.

Reply:
I thought we agreed to do away with magic?

Reply:
Everywhere but the proxy generators. Im not sure how to implement the Modifiers without them proxying calls to Generator. 

I would be super happy to get a suggestion.

Reply:
Isn't the `UniqueGenerator` just an implementation of a generator, hence returning `self` would be true, since it extends generator?

Reply:
No not really. It is a proxy. It does not extend.